src/main/java/org/ngafid/uploads/process/format/ParquetFileProcessor.java

```diff
@@ -0,0 +1,249 @@
+package org.ngafid.uploads.process.format;
+
+import org.apache.avro.generic.GenericRecord;
+import org.apache.parquet.avro.AvroParquetReader;
+import org.apache.parquet.hadoop.ParquetReader;
+import org.apache.parquet.io.InputFile;
+import org.jline.utils.Log;
+import org.ngafid.common.MD5;
+import org.ngafid.common.TimeUtils;
+import org.ngafid.flights.Airframes;
+import org.ngafid.flights.DoubleTimeSeries;
+import org.ngafid.flights.Parameters;
+import org.ngafid.flights.StringTimeSeries;
+import org.ngafid.uploads.process.FatalFlightFileException;
+import org.ngafid.uploads.process.FlightMeta;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.util.*;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+/**
+ * Processes a Parquet file and extracts flights.
+ */
+public class ParquetFileProcessor {
+    private static final Logger LOG = Logger.getLogger(ParquetFileProcessor.class.getName());
+    private final Connection connection;
+    private final InputFile inputFile;
+    private final String filename;
+
+    public ParquetFileProcessor(Connection connection, InputFile inputFile, String filename) {
+        this.connection = connection;
+        this.inputFile = inputFile;
+        this.filename = filename;
+    }
+
+    public Stream<FlightBuilder> parse() {
+        LOG.info("Parsing Parquet file: " + filename);
+        List<FlightBuilder> flightBuilders = new ArrayList<>();
+
+        try (ParquetReader<GenericRecord> reader = AvroParquetReader.<GenericRecord>builder(inputFile).build()) {
+            GenericRecord record;
+
+           int counter = 0;
+            while ((record = reader.read()) != null) {
+                FlightMeta flightMeta = new FlightMeta();
+
+                Log.info("Processing metadata for flight " + counter);
+                processMetaData(record, flightMeta);
+
+                Map<String, DoubleTimeSeries> doubleTimeSeries = new HashMap<>();
+                Map<String, StringTimeSeries> stringTimeSeries = new HashMap<>();
+
+                populateTimeSeries(record, doubleTimeSeries, stringTimeSeries);
+
+                int numberRows = doubleTimeSeries.get(Parameters.UNIX_TIME_SECONDS).size();
+                LOG.info("Number of rows for flight " + counter + ": " + numberRows);
+
+
+                flightBuilders.add(new FlightBuilder(flightMeta, doubleTimeSeries, stringTimeSeries));
```

Comment:
As per comments above, you will need to create a `ParquetFlightBuilder` subclass of `FlightBuilder`